iree/hal/local/loaders/legacy_library_loader.cc

```diff
@@ -81,6 +81,13 @@ static iree_status_t iree_hal_dylib_executable_flatbuffer_verify(
                             "executable library_embedded is missing/empty");
   }
 
+  if (iree_DyLibExecutableDef_sanitized_kind_get(executable_def) ==
+          iree_Sanitizer_kAddress &&
+      !__has_feature(address_sanitizer)) {
+    return iree_make_status(IREE_STATUS_UNAVAILABLE,
+                            "Failed to load ASAN libraries");
```

Comment:
to help users who encounter this it's useful to hint at the precise problem and what to do about it, like: `"dynamic library executable is compiled with ASAN support but the host application is not; enable ASAN to load this executable"`